### PR TITLE
CLI Updates: Restore VT100 support, help, etc.

### PR DESCRIPTION
**Pull request by penrods**
## Description

CLI Updates: Restore VT100 support, help, etc.
* Restored VT100 function key support
* ESC now clears the entry line
* Spoken commands now become part of the history
* Enhanced help screen with auto word-wrap
* BUG: "Show meter" setting didn't persist

## How to test
Fire up the CLI and...
* Resize a SSH window while in :help 
* Change meter display, restart CLI and verify
* Type stuff and hit ESC, will clear
* Speak commands and see in history window / Ctrl+N, Ctrl+P history

## Contributor license agreement signed?
CLA [X ] (Whether you have signed a [CLA - Contributor Licensing Agreement](https://mycroft.ai/cla/)

### CODE REVIEW

File: mycroft/client/text/main.py [hunk 1]
```diff
@@ -742,42 +780,72 @@ def num_help_pages():
     return ceil(lines / (curses.LINES - 4))
 
 
-def show_help():
-    global scr
-    global screen_mode
+def __render_help(scr, txt, y_pos, i, first_line, last_line, clr):
```
[the remark this answers is not in the document]
Comment (penrods) [reply mycroft/client/text/main.py]: Yeah, that makes sense.  I was just using "__" to distinguish "helpers", but not really a big deal inside a self-contained program.

File: mycroft/client/text/main.py [hunk 2]
```diff
@@ -742,42 +780,72 @@ def num_help_pages():
...
 
-    help_new_page()
-    y_pos = 2
-    page = 1
+def _do_draw_help(scr):
+    _help_draw_header(scr)
+    y = 2
+    page = subscreen+1
+
+    first = subscreen * (curses.LINES-4)  # account for header
```
[the remark this answers is not in the document]
Comment (penrods) [reply mycroft/client/text/main.py]: Yep!  Nice catch.

File: mycroft/client/text/main.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (penrods) [reply mycroft/client/text/main.py]: I ended up doing a different reorg, making render_footer(), etc. and making those nested functions.  I think it comes out cleaner in this case, too.

File: mycroft/client/text/main.py [hunk 3]
```diff
@@ -686,53 +702,75 @@ def _do_drawing(scr):
...
+help_longest = 0
+for s in help_struct:
+    for ent in s[1]:
+        help_longest = max(help_longest, len(ent[0]))
 
 
-def help_new_page():
+def _help_draw_header(scr):
     # Header
     scr.erase()
```
[the remark this answers is not in the document]
Comment (penrods) [reply mycroft/client/text/main.py]: See above

**Comment by forslund** [comment]
Looks good, Merging

**Pull request merged**